src/main/scala/nl.knaw.dans.easy.deposit/DepositDir.scala

```diff
@@ -75,21 +75,23 @@ case class DepositDir private(baseDir: File, user: String, id: UUID) extends Deb
    */
   def setStateInfo(stateInfo: StateInfo): Try[Unit] = {
     for {
-      props <- getDepositProps
-      currentState = State.withName(props.getString("state.label"))
-      _ <- checkStateTransition(currentState, stateInfo.state)
+      props <- checkStateTransition(stateInfo.state) // read again as submit might have added a DOI
```

Comment:
isn't it a bit weird to get the props returned for this check? I liked the previous def better, where there were two state-parameters

Reply:
I agree, it is almost always best [not to mix queries and commands](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation).